src/pudl/workspace/setup.py

```diff
@@ -58,139 +46,38 @@ def get_defaults(
-
-    return yaml_settings
-
-
-def set_defaults(pudl_in, pudl_out, clobber=False):
-    """Set default user input and output locations in ``$HOME/.pudl.yml``.
-
-    Create a user settings file for future reference, that defines the default
-    PUDL input and output directories. If this file already exists, behavior
-    depends on the clobber parameter, which is False by default. If it's True,
-    the existing file is replaced. If False, the existing file is not changed.
-
-    Args:
-        pudl_in (os.PathLike): Path to be used as the default input directory
-            for PUDL -- this is where :mod:`pudl.workspace.datastore` will look
-            to find the ``data`` directory, full of data from public agencies.
-        pudl_out (os.PathLike): Path to the default output directory for PUDL,
-            where results of data processing will be organized.
-        clobber (bool): If True and a user settings file exists, overwrite it.
-            If False, do not alter the existing file. Defaults to False.
-
-    Returns:
-        None
-    """
-    logger.warning(
-        "pudl_settings is being deprecated in favor of environment "
-        "variables PUDL_OUTPUT and PUDL_INPUT. For more info "
-        "see: https://catalystcoop-pudl.readthedocs.io/en/dev/dev/dev_setup.html"
-    )
-    settings_file = pathlib.Path.home() / ".pudl.yml"
-    if settings_file.exists():
-        if clobber:
-            logger.info(f"{settings_file} exists: clobbering.")
-        else:
-            logger.info(f"{settings_file} exists: not clobbering.")
-            return
-
-    with settings_file.open(mode="w") as f:
-        f.write(f"pudl_in: {pudl_in.expanduser().resolve()}\n")
-        f.write(f"pudl_out: {pudl_out.expanduser().resolve()}\n")
-
-
-def derive_paths(pudl_in, pudl_out):
-    """Derive PUDL paths based on given input and output paths.
-
-    If no configuration file path is provided, attempt to read in the user
-    configuration from a file called .pudl.yml in the user's HOME directory.
-    Presently the only values we expect are pudl_in and pudl_out, directories
-    that store files that PUDL either depends on that rely on PUDL.
+def derive_paths(pudl_in: Path, pudl_out: Path) -> dict[str, str]:
+    """Derive PUDL paths based on given input and output env variables.
 
     Args:
-        pudl_in (os.PathLike): Path to the directory containing the PUDL input
-            files, most notably the ``data`` directory which houses the raw
-            data downloaded from public agencies by the
-            :mod:`pudl.workspace.datastore` tools. ``pudl_in`` may be the same
-            directory as ``pudl_out``.
-        pudl_out (os.PathLike): Path to the directory where PUDL should write
-            the outputs it generates. These will be organized into directories
-            according to the output format (sqlite, parquet, etc.).
+        pudl_in (Path): directory containing PUDL input files, most notably
```

Comment:
I think this is now the `data` directory itself, not the parent folder.

Reply:
Yeah, the non-consistent semantics for input/output env variables is somewhat strange and counter-intuitive. I was caught off guard by this few times.